python-sdk/tutorials/automl-with-azureml/ONNX_batch_model_generator_automl_for_images.py

```diff
@@ -0,0 +1,254 @@
+                      input_names=input_names,
+                      output_names=output_names,
+                      dynamic_axes=dynamic_axes)
+
+
+def generate_onnx_batch_model(batch_size, height_onnx, width_onnx, task_type, best_child_run,
+                              model_settings, model_name, device):
+    """
+    Get the onnx batch model and verify the schema.
+
+    :param batch_size: Batch size needed for batch inference
+    :type batch_size: int
+    :param height_onnx: Height of the image to be used for ONNX model
+    :type height_onnx: int
+    :param width_onnx: Width of the image to be used for ONNX model
+    :type width_onnx: int
+    :param task_type: Type of vision task from image classification, object detection, segmentation
+    :type task_type: str
+    :param best_child_run: Best child run object of an experiment
+    :type best_child_run: azureml.core.run.Run
+    :param model_settings: Model parameters with key, values
+    :type model_settings: dict
+    :param model_type: Model name
+    :type model_type: str
+    :param device: Device on which the model has to be generated
+    :type device: str
+    """
+
+    # download the pytorch model weights
+    best_child_run.download_file(name=os.path.join(ArtifactLiterals.OUTPUT_DIR, shared_constants.PT_MODEL_FILENAME),
+                                 output_file_path=shared_constants.PT_MODEL_FILENAME)
+    # load the model wrapper
+    model_wrapper = load_model(task_type, shared_constants.PT_MODEL_FILENAME, **model_settings)
+    onnx_model_file_path = './outputs/model_' + str(batch_size) + '.onnx'
+
+    if batch_size <= 1:
+        msg = 'Please use the auto-generated ONNX model for the best child run. No need to run this script'
+        logger.warning(msg)
+        return
+    
+    input_names = ['input']
+    dummy_input = torch.randn(batch_size, 3, height_onnx, width_onnx).to(device)
+
+
+    if (
+        ModelNames.FASTER_RCNN_RESNET18_FPN == model_name
+        or ModelNames.FASTER_RCNN_RESNET34_FPN == model_name
+        or ModelNames.FASTER_RCNN_RESNET50_FPN == model_name
+        or ModelNames.FASTER_RCNN_RESNET101_FPN == model_name
+        or ModelNames.FASTER_RCNN_RESNET152_FPN == model_name
+        or ModelNames.RETINANET_RESNET50_FPN == model_name
+    ):
+
+        if ModelNames.RETINANET_RESNET50_FPN == model_name:
+            od_output_names = ['boxes', 'scores', 'labels']
+        else:
+            od_output_names = ['boxes', 'labels', 'scores']
+        output_names = [name + "_" + str(sample_id) for sample_id in range(batch_size) for name in od_output_names]
+        dynamic_axes = dict()
+        dynamic_axes['input'] = {0: 'batch', 1: 'channel', 2: 'height', 3: 'width'}
+        for output in output_names:
+            dynamic_axes[output] = {0: 'prediction'}
+
+        model_wrapper.disable_model_transform()
+        model = model_wrapper.model
+
+    elif ModelNames.YOLO_V5 == model_name:
```

Comment:
Where is the post-processing part for yolo like in the notebook?

Reply:
this is not integrated into the onnx now. This requires many changes and I will look into this later!

Reply:
no..I didn't mean to add post-processing as part of ONNX. I meant the original part for post-processing which is in the notebook to be used after generating onnx model. 
My question is that even after we generate onnx using this file, we still need to use your notebook for actual inference? What is the workflow?

Reply:
yes. I have added it to doc! https://review.docs.microsoft.com/en-us/azure/machine-learning/how-to-inference-onnx-automl-image-models?branch=pr-en-us-185269&tabs=object-detect-yolo#postprocessing